src/test/java/StringManipulationTest.java

```diff
@@ -87,27 +130,66 @@ public void testGeSubStrings1() {
         assertEquals(sStings[1], "string");
     }
 
+    /**
+     * Check whether an illegal argument exception is correctly thrown when one of the string indices is zero.
+     */
     @Test
     public void testGeSubStrings2() {
-        fail("Not yet implemented");
+        manipulatedstring.setString("a b c d");
+        assertThrows(IllegalArgumentException.class, () -> manipulatedstring.getSubStrings(0, 1));
     }
+
+    /**
+     * Check whether an illegal argument exception is correctly thrown when the second index is lower than the first.
+     */
     @Test
     public void testGeSubStrings3() {
-        fail("Not yet implemented");
+        manipulatedstring.setString("a b c d");
+        assertThrows(IllegalArgumentException.class, () -> manipulatedstring.getSubStrings(2, 1));
     }
+    /**
+     * Check whether an out of bounds exception is correctly thrown when the right index is higher than the number of
+     * words.
+     */
     @Test
     public void testGeSubStrings4() {
-        fail("Not yet implemented");
+        manipulatedstring.setString("a b c d");
+        assertThrows(IndexOutOfBoundsException.class, () -> manipulatedstring.getSubStrings(2, 5));
     }
+
+    /**
+     * Check whether a single string is correctly returned when the indices are the same and there is only
+     * one word.
+     */
     @Test
     public void testGeSubStrings5() {
-        fail("Not yet implemented");
+        manipulatedstring.setString("abc");
+        assertEquals("abc", manipulatedstring.getSubStrings(1, 1)[0]);
     }
+
+    /**
+     * Check whether an out of bounds exception is correctly thrown for a string of whitespace with nonzero indices.
+     */
     @Test
     public void testGeSubStrings6() {
-        fail("Not yet implemented");
+        manipulatedstring.setString("   ");
+        assertThrows(IndexOutOfBoundsException.class, () -> manipulatedstring.getSubStrings(1, 1));
     }
 
+    /**
+     * Check whether a simple string with words separated by multiple spaces is correctly converted to substrings.
+     */
+    @Test
+    public void testGeSubStrings7() {
+        manipulatedstring.setString("This    is    my   string");
+        String [] sStings = manipulatedstring.getSubStrings(3, 4);
+        assertEquals(sStings[0], "my");
+        assertEquals(sStings[1], "string");
+    }
+
```

Comment:
This is a good test case. Have you considered what might happen if the string contains other types of whitespace characters, like tabs or newlines?